src/System/System.php

```diff
@@ -128,4 +157,360 @@ static public function isArch(string $arch): bool
+                $data['total']['steal'] += $cpu['steal'];
+                $data['total']['guest'] += $cpu['guest'];
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Gets the current usage of a core as a percentage. Passing 0 will return the usage of all cores combined.
+     * 
+     * @param int $core
+     * 
+     * @return int
+     * 
+     * @throws Exception
+     */
+    static public function getCPUUtilisation(int $id = 0): int
+    {
+        switch (self::getOS()) {
+            case 'Linux':
+                $cpuNow = self::getProcStatData();
+                $i = 0;
+
+                $data = [];
+
+                foreach ($cpuNow as $cpu) {
+                    // Check if this is the total CPU
+                    $cpuTotal = $cpu['user'] + $cpu['nice'] + $cpu['system'] + $cpu['idle'] + $cpu['iowait'] + $cpu['irq'] + $cpu['softirq'] + $cpu['steal'];
+
+                    $cpuIdle = $cpu['idle'];
+
+                    $idleDelta = $cpuIdle - (isset($lastData[$i]) ? $lastData[$i]['idle'] : 0);
+
+                    $totalDelta = $cpuTotal - (isset($lastData[$i]) ? $lastData[$i]['total'] : 0);
+
+                    $lastData[$i]['total'] = $cpuTotal;
+                    $lastData[$i]['idle'] = $cpuIdle;
+
+                    $result = (1.0 - ($idleDelta / $totalDelta)) * 100;
+
+                    $data[$i] = $result;
+
+                    $i++;
+                }
+
+                if ($id === 0) {
+                    return intval(array_sum($data));
+                } else {
+                    return $data[$id];
+                }
+            default:
+                throw new Exception(self::getOS() . " not supported.");
+        }
+    }
+
+    /**
+     * Returns the total amount of RAM available on the system as Megabytes.
+     * 
+     * @return int
+     * 
+     * @throws Exception
+     */
+    static public function getMemoryTotal(): int
+    {
+        switch (self::getOS()) {
+            case 'Linux':
+                $meminfo = file_get_contents('/proc/meminfo');
+                preg_match('/MemTotal:\s+(\d+)/', $meminfo, $matches);
+                return intval(intval($matches[1]) / 1024);
+                break;
+            case 'Darwin':
+                return intval((intval(shell_exec('sysctl -n hw.memsize'))) / 1024 / 1024);
+                break;
+            default:
+                throw new Exception(self::getOS() . " not supported.");
+            }
+    }
+
+    /**
+     * Returns the total amount of Free RAM available on the system as Megabytes.
+     * 
+     * @return int
+     * 
+     * @throws Exception
+     */
+    static public function getMemoryFree(): int
+    {
+        switch (self::getOS()) {
+            case 'Linux':
+                $meminfo = file_get_contents('/proc/meminfo');
+                preg_match('/MemFree:\s+(\d+)/', $meminfo, $matches);
+                return intval($matches[1] / 1024);
```

Comment:
Accessing the array key directly without validating it exists is risky. Let's have a null safety check.